state/txindex/kv/kv_test.go

```diff
@@ -132,6 +135,73 @@ func TestTxSearch(t *testing.T) {
 	}
 }
 
+func TestTxSearchWithCancelation(t *testing.T) {
+	allowedKeys := []string{"account.number", "account.owner", "account.date"}
+	indexer := NewTxIndex(db.NewMemDB(), IndexEvents(allowedKeys))
+
+	txResult := txResultWithEvents([]abci.Event{
+		{Type: "account", Attributes: []kv.Pair{{Key: []byte("number"), Value: []byte("1")}}},
+		{Type: "account", Attributes: []kv.Pair{{Key: []byte("owner"), Value: []byte("Ivan")}}},
+		{Type: "", Attributes: []kv.Pair{{Key: []byte("not_allowed"), Value: []byte("Vlad")}}},
+	})
+	hash := txResult.Tx.Hash()
+
+	err := indexer.Index(txResult)
+	require.NoError(t, err)
+
+	testCases := []struct {
+		q             string
+		resultsLength int
+	}{
+		// search by hash
+		{fmt.Sprintf("tx.hash = '%X'", hash), 0},
+		// search by exact match (one key)
+		{"account.number = 1", 0},
+		// search by exact match (two keys)
+		{"account.number = 1 AND account.owner = 'Ivan'", 0},
+		// search by exact match (two keys)
+		{"account.number = 1 AND account.owner = 'Vlad'", 0},
+		{"account.owner = 'Vlad' AND account.number = 1", 0},
+		{"account.number >= 1 AND account.owner = 'Vlad'", 0},
+		{"account.owner = 'Vlad' AND account.number >= 1", 0},
+		{"account.number <= 0", 0},
+		{"account.number <= 0 AND account.owner = 'Ivan'", 0},
+		// search using a prefix of the stored value
+		{"account.owner = 'Iv'", 0},
+		// search by range
+		{"account.number >= 1 AND account.number <= 5", 0},
+		// search by range (lower bound)
+		{"account.number >= 1", 0},
+		// search by range (upper bound)
+		{"account.number <= 5", 0},
+		// search using not allowed key
+		{"not_allowed = 'boom'", 0},
+		// search for not existing tx result
+		{"account.number >= 2 AND account.number <= 5", 0},
+		// search using not existing key
+		{"account.date >= TIME 2013-05-03T14:45:00Z", 0},
+		// search using CONTAINS
+		{"account.owner CONTAINS 'an'", 0},
+		// search for non existing value using CONTAINS
+		{"account.owner CONTAINS 'Vlad'", 0},
+		// search using the wrong key (of numeric type) using CONTAINS
+		{"account.number CONTAINS 'Iv'", 0},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	cancel()
```

Comment:
By the looks of it, we only test the very first termination clause here, not others. My question then is why do we need all the queries above? Isn't one sufficient?

Reply:
Perhaps, I just copy/pasted the queries from the other test, setting the expected results to 0 in all cases. Again I think it would be very hard to test specific timeout scenarios.